tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -464,6 +464,99 @@ class SerializableModelLibraryGenerator {
     });
   }
 
+  Expression _buildClonableCopyWithAssignment(
+      SerializableModelFieldDefinition field) {
+    var shouldDeepClone = field.type.isListType || field.type.isMapType;
+
+    if (!shouldDeepClone) {
+      return refer(cloneTrivialFunctionName, serverpodUrl(serverCode))
+          .call([refer(Keyword.thisKeyword).property(field.name)]);
+    }
+
+    return _buildDeepCloneTree(
+      field.type,
+      field.name,
+      root: true,
+    );
+  }
+
+  Expression _buildDeepCloneTree(TypeDefinition type, String variableName,
+      {int depth = 0, bool root = false}) {
+    var isLeafNode = type.generics.isEmpty;
+    if (isLeafNode) {
+      return refer(cloneTrivialFunctionName, serverpodUrl(serverCode))
+          .call([refer(variableName)]);
+    }
+
+    var nextCallback = switch (type.className) {
+      ListKeyword.className =>
+        _buildListCloneCallback(type.generics.first, depth),
+      MapKeyword.className =>
+        _buildMapCloneCallback(type.generics[0], type.generics[1], depth),
+      _ => throw "Can't clone type ${type.className}",
```

Comment:
What's the convention when throwing stuff? Is it better to always wrap it in an Exception type?

Reply:
Yes, we usually try to use a defined exception type instead of throwing just a `String`.

There are a couple of Exceptions defiend that can be handy along with Errors if the state is unrecourable.
I would say an error is more in line with the code here since we always want the generator to stop if this occurs.

Reply:
Will throw `UnimplementedError` instead